### PR TITLE
feat: Add support for Dart `3.1`, `3.2`, `3.3`; remove support for Dart `2.19`

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue

Closes: #992

## Approach
n/a

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

File: packages/dart/lib/src/base/parse_constants.dart [hunk 1]
```diff
@@ -1,4 +1,4 @@
-part of flutter_parse_sdk;
+part of '../../parse_server_sdk.dart';
```
Comment (mtrezza) [inline packages/dart/lib/src/base/parse_constants.dart:1]: Why this change?

**Comment by mtrezza** [comment]
This is a breaking change since we are removing support for older dart versions, right? Could you please rename the PR title to

`feat: Add support for Dart x.x, x.x; remove support for Dart x.x, x.x`

File: packages/dart/lib/src/base/parse_constants.dart (hunk 1 above)
Comment (mbfakourii) [reply packages/dart/lib/src/base/parse_constants.dart:1]: It is recommended to use this method in new versions of [Lint](https://pub.dev/packages/flutter_lints).
If we don't use it, we will get a warning and we will have problems in ci !

**Comment by mbfakourii** [comment]
> This is a breaking change since we are removing support for older dart versions, right? Could you please rename the PR title to
> 
> `feat: Add support for Dart x.x, x.x; remove support for Dart x.x, x.x`

Yes, the title has also changed.

File: packages/dart/lib/src/base/parse_constants.dart [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 
 // Library
 const String keySdkVersion = '6.4.0';
```
Comment (mtrezza) [inline packages/dart/lib/src/base/parse_constants.dart:4]: This needs to be version 7.0.0, right?

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
Could you add the changelog entry?

```
Features:
- ...

Breaking Changes:
- Removes support for Dart ...
```

**Comment by mbfakourii** [comment]
> Could you add the changelog entry?
> 
> ```
> Features:
> - ...
> 
> Breaking Changes:
> - Removes support for Dart ...
> ```

Added.

**Comment by mbfakourii** [comment]
CI errors are related to `codecov`.

**Review by mtrezza — APPROVED** [review]

File: packages/dart/lib/src/utils/parse_live_list.dart [hunk 3]
```diff
@@ -539,6 +540,10 @@ class ParseLiveElement<T extends ParseObject> extends ParseLiveListElement<T> {
               super.object = parseResponse.result.first;
             }
             break;
+          case LiveQueryClientEvent.disconnected:
+            break;
+          case LiveQueryClientEvent.userDisconnected:
+            break;
```
Comment (mtrezza) [inline packages/dart/lib/src/utils/parse_live_list.dart:546]: Just wondering, these added cases do not look like mere refactors. What is that change? If it adds or changes a feature, then it probably needs its own PR.

Comment (mbfakourii) [reply packages/dart/lib/src/utils/parse_live_list.dart:546]: If we do not consider these `case's in the new version of lint, lint will consider them as warnings and we will face problems in CI.

Comment (mtrezza) [reply packages/dart/lib/src/utils/parse_live_list.dart:546]: Got it

File: packages/dart/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,13 @@
+## [7.0.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.4.0...dart-7.0.0) (2024-04-11)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [7.0.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.4.0...dart-7.0.0) (2024-04-12)
```

**Review by mtrezza — CHANGES_REQUESTED** [review]
